File: src/1267-Count-Servers-that-Communicate/1267.java
class Solution 
{
public:
    int countServers(vector<vector<int>>& grid) 
    {
        int m = grid.size(), n = grid[0].size(), res = 0;
        int r[m] = {}, c[n] = {};
        for (int i = 0; i < m; i++) 
        {
            for (int j = 0; j < n; j++)
            {
                if (grid[i][j])
                {
                    r[i]++; c[j]++;
                }
            }
        }
        
        for (int i = 0; i < m; i++)
        {
            for (int j = 0; j < n; j++)
            {
                if (grid[i][j] && (r[i] + c[j] > 2)) res++;
            }
        }
        return res;
    }
};

File: src/1267-Count-Servers-that-Communicate/1267.js
var countServers = function(grid) {
    let m = grid.length, n = grid[0].length, res = 0;
    let r = new Array(m), c = new Array(n);
    r.fill(0), c.fill(0);
    
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            if (grid[i][j]) {
                r[i]++; c[j]++;
            }
        }
    }
    
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            if (grid[i][j] && (r[i] + c[j] > 2)) 
                res++;
        }
    }
    return res;
};